app/models/name/taxonomy.rb

```diff
@@ -516,14 +519,16 @@ def propagate_classification
   # This is meant to be run nightly to ensure that all the classification
   # caches are up to date.  It only pays attention to genera or higher.
   def self.refresh_classification_caches
-    Name.connection.execute(%(
+    statement = ActiveRecord::Base.connection.raw_connection.prepare(%(
       UPDATE names n, name_descriptions nd
       SET n.classification = nd.classification
       WHERE nd.id = n.description_id
-        AND n.rank <= #{Name.ranks[:Genus]}
+        AND n.`rank` <= ?
         AND nd.classification != n.classification
         AND COALESCE(nd.classification, "") != ""
     ))
+    statement.execute(Name.ranks[:Genus])
+    statement.close
```

Comment:
Questions
- **Is this right?**
- **Is there a better/more idiomatic/less specific way to do this?**

I copied this -- without understanding `ActiveRecord::Base.connection.raw_connection` -- from https://stackoverflow.com/questions/4483049/how-to-execute-a-raw-update-sql-with-dynamic-binding-in-rails.
That part feels too specific, sort of a Liskov-type issue.
Is there a better way to do it? Perhaps with [`exec_update`](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/DatabaseStatements.html#method-i-exec_update)?

Reply:
Why do we need to do this at all?  Be careful with these cache updates.  I'd recommend avoiding stylistic revisions unless there is a compelling reason to do so.